cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -220,12 +223,25 @@ protected void addPartitionToExplore(PartitionKey key, String path) {
   @Override
   public void dropPartition(PartitionKey key) {
     byte[] rowKey = generateRowKey(key, partitioning);
+    Partition partition = getPartition(key);
+    if (partition == null) {
+      // silently ignore non-existing partitions
```

Comment:
We have always silently ignored it (by deleting a non-existent row). I only added comment to make that clear.